core/src/mindustry/entities/Effect.java

```diff
@@ -52,6 +52,17 @@ public Effect layer(float l, float duration){
         this.layerDuration = duration;
         return this;
     }
+    
+    public Effect ground(){
+        layer = Layer.debris;
+        return this;
+    }
+    
+    public Effect ground(float duration){
+        layer = Layer.debris;
+        this.layerDuration = duration;
```

Comment:
```suggestion
        layerDuration = duration;
```

Reply:
I don't know, the original code before *and* after #3184 has the "this.".

Reply:
There is to difference between them, just without `this` is shorter.

Reply:
`this` is used where the input value and the variable are called the same thing.
If it were `ground(float layerDuration)`, then you would have to do `this.layerDuration = layerDuration` for it to work properly.
I took a 2 week basic Java camp during the summer.

Reply:
since input `duration` is not named `layerDuration` the `this.` is redundant

Reply:
then what about the functions right above it? the layer() of line 45.